packages/x-components/src/x-modules/experience-controls/store/__tests__/actions.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { getFetchMock, installNewXPlugin } from '../../../../__tests__/utils';
+import { fetchExperienceControlsResponse } from '../actions/fetch-controls.action';
+import { createExperienceControlsStore, resetExperienceControlsStateWith } from './utils';
+
+describe('testing experience controls module actions', () => {
+  const mockedControls = { numberOfCarousels: 10, resultsPerCarousels: 21 };
```

Comment:
The endpoint returns controls only, but the adapter will generate the events. Here you're mocking the adapter response, so it should have events too.